test/hpke_test.c

```diff
@@ -0,0 +1,1804 @@
+
+                        res = (overallresult == 1 ? "worked" : "failed");
+                        TEST_note("HPKE %s for mode: %s/0x%02x, "\
+                                  "kem: %s/0x%02x, kdf: %s/0x%02x, "\
+                                  "aead: %s/0x%02x",
+                                  res, mode_str_list[mind], (int) mind,
+                                  kem_str_list[kemind], kem_id,
+                                  kdf_str_list[kdfind], kdf_id,
+                                  aead_str_list[aeadind], aead_id);
+                    }
+                }
+            }
+            EVP_PKEY_free(authpriv);
+        }
+    }
+    return overallresult;
+}
+
+/**
+ * @brief check roundtrip for export
+ * @return 1 for success, other otherwise
+ */
+static int test_hpke_export(void)
+{
+    EVP_PKEY *privp = NULL;
+    unsigned char pub[OSSL_HPKE_TSTSIZE];
+    size_t publen = sizeof(pub);
+    int hpke_mode = OSSL_HPKE_MODE_BASE;
+    OSSL_HPKE_SUITE hpke_suite = OSSL_HPKE_SUITE_DEFAULT;
+    OSSL_HPKE_CTX *ctx = NULL;
+    OSSL_HPKE_CTX *rctx = NULL;
+    unsigned char exp[32];
+    unsigned char exp2[32];
+    unsigned char rexp[32];
+    unsigned char rexp2[32];
+    unsigned char plain[] = "quick brown fox";
+    size_t plainlen = sizeof(plain);
+    unsigned char enc[OSSL_HPKE_TSTSIZE];
+    size_t enclen = sizeof(enc);
+    unsigned char cipher[OSSL_HPKE_TSTSIZE];
+    size_t cipherlen = sizeof(cipher);
+    unsigned char clear[OSSL_HPKE_TSTSIZE];
+    size_t clearlen = sizeof(clear);
+    char *estr = "foo";
+
+    if (!TEST_true(OSSL_HPKE_keygen(hpke_suite, pub, &publen, &privp,
+                                    NULL, 0, testctx, NULL)))
+        goto end;
+    if (!TEST_ptr(ctx = OSSL_HPKE_CTX_new(hpke_mode, hpke_suite,
+                                          testctx, NULL)))
+        goto end;
+    /* a few error cases 1st */
+    if (!TEST_false(OSSL_HPKE_export(NULL, exp, sizeof(exp),
+                                     (unsigned char *)estr, strlen(estr))))
+        goto end;
+    /* ctx before encap should fail too */
+    if (!TEST_false(OSSL_HPKE_export(ctx, exp, sizeof(exp),
+                                     (unsigned char *)estr, strlen(estr))))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_encap(ctx, enc, &enclen, pub, publen, NULL, 0)))
+        goto end;
+    if (!TEST_true(OSSL_HPKE_seal(ctx, cipher, &cipherlen, NULL, 0,
+                                  plain, plainlen)))
+        goto end;
+    /* now for real */
+    if (!TEST_true(OSSL_HPKE_export(ctx, exp, sizeof(exp),
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    /* check a 2nd call with same input gives same output */
+    if (!TEST_true(OSSL_HPKE_export(ctx, exp2, sizeof(exp2),
+                                    (unsigned char *)estr, strlen(estr))))
+        goto end;
+    if (!TEST_true(TEST_mem_eq(exp, sizeof(exp), exp2, sizeof(exp2))))
```

Comment:
You don't need to wrap `TEST_mem_eq` inside `TEST_true`

Reply:
did that in various places